x-pack/platform/plugins/shared/fleet/server/tasks/sync_integrations/compare_synced_integrations.ts

```diff
@@ -0,0 +1,295 @@
+        return {
+          ...ccrIntegration,
+          sync_status: 'failed' as SyncStatus.FAILED,
+          error: `Found incorrect installed version ${localIntegrationSO?.attributes.version}`,
+        };
+      }
+      if (localIntegrationSO?.attributes.install_status !== 'installed') {
+        return {
+          ...ccrIntegration,
+          sync_status: 'failed' as SyncStatus.FAILED,
+          error: `Installation status: ${localIntegrationSO?.attributes.install_status}`,
+        };
+      }
+      return {
+        ...ccrIntegration,
+        sync_status: 'completed' as SyncStatus.COMPLETED,
+        updated_at: localIntegrationSO?.updated_at,
+      };
+    }
+  );
+  return { integrations: integrationsStatus };
+};
+
+const fetchAndCompareCustomAssets = async (
+  esClient: ElasticsearchClient,
+  logger: Logger,
+  ccrCustomAssets: { [key: string]: CustomAssetsData }
+): Promise<RemoteSyncedCustomAssetsRecord | undefined> => {
+  if (!ccrCustomAssets) return;
+
+  const abortController = new AbortController();
+
+  try {
+    const installedPipelines = await getPipeline(esClient, CUSTOM_ASSETS_PREFIX, abortController);
+    const installedComponentTemplates = await getComponentTemplate(
+      esClient,
+      CUSTOM_ASSETS_PREFIX,
+      abortController
+    );
+
+    const componentTemplatesByName = (
+      installedComponentTemplates?.component_templates || []
+    ).reduce((acc, componentTemplate) => {
+      if (componentTemplate?.name) {
+        acc[componentTemplate.name] = componentTemplate.component_template?.template;
+      }
+      return acc;
+    }, {} as Record<string, ClusterComponentTemplateSummary>);
+    const componentTemplates = installedComponentTemplates?.component_templates
+      ? componentTemplatesByName
+      : undefined;
+    const result: RemoteSyncedCustomAssetsRecord = {};
+
+    // compare custom pipelines and custom component templates
+    Object.entries(ccrCustomAssets).forEach(([ccrCustomName, ccrCustomAsset]) => {
+      const res = compareCustomAssets({
+        ccrCustomAsset,
+        ingestPipelines: installedPipelines,
+        componentTemplates,
+      });
+      result[ccrCustomName] = res;
+    });
+    return result;
+  } catch (error) {
+    logger.error('error', error?.message);
+    return { error: error?.message };
+  }
+};
+
+const compareCustomAssets = ({
+  ccrCustomAsset,
+  ingestPipelines,
+  componentTemplates,
+}: {
+  ccrCustomAsset: CustomAssetsData;
+  ingestPipelines?: IngestGetPipelineResponse;
+  componentTemplates?: Record<string, ClusterComponentTemplateSummary>;
+}): RemoteSyncedCustomAssetsStatus => {
+  if (ccrCustomAsset.type === 'ingest_pipeline') {
+    if (!ingestPipelines) {
+      return {
+        ...ccrCustomAsset,
+        sync_status: 'synchronizing' as SyncStatus.SYNCHRONIZING,
+      };
+    }
+
+    const installedAsset = ingestPipelines[ccrCustomAsset?.name];
+
+    if (isEqual(installedAsset, ccrCustomAsset?.pipeline)) {
```

Comment:
we still miss the version comparison here

Reply:
Added!